shell/platform/android/platform_view_android_jni.cc

```diff
@@ -11,7 +11,19 @@
 #include "lib/ftl/arraysize.h"
 #include "lib/ftl/logging.h"
 
-#define PLATFORM_VIEW reinterpret_cast<PlatformViewAndroid*>(platform_view)
+template <typename T>
+class HeapSharedPtr {
```

Comment:
Trough the JNI we can move just Java Objects or Atomic values.
The `std::shared_ptr` cannot be passed directly, but the `std::make_shared` method will create it on the stack.
We wrap it into an object in the heap that will be manually managed.